oidc.go

```diff
@@ -307,3 +313,16 @@ func (j *jsonTime) UnmarshalJSON(b []byte) error {
 	*j = jsonTime(time.Unix(unix, 0))
 	return nil
 }
+
+func unmarshalResp(r *http.Response, body []byte, v interface{}) error {
+	err := json.Unmarshal(body, &v)
+	if err != nil {
+		ct := r.Header.Get("Content-Type")
+		mediaType, _, _ := mime.ParseMediaType(ct)
```

Comment:
@ericchiang The extra condition is unnecessary though, it's the same logic with, or without it.

We don't need to check the error because if the `err != nil`, then `mediaType != "application/json"` (because `mediaType == ""` in error cases, so the outcome is the same. I don't feel too strongly about it, I can add it if you like, but it just seems like unnecessary extra noise.